freqtrade/vendor/qtpylib/indicators.py

```diff
@@ -103,47 +104,50 @@ def session(df, start='17:00', end='16:00'):
 
     return df.copy()
 
-
 # ---------------------------------------------
 
+
 def heikinashi(bars):
     bars = bars.copy()
     bars['ha_close'] = (bars['open'] + bars['high'] +
                         bars['low'] + bars['close']) / 4
 
-    bars['ha_open'] = (bars['open'].shift(1) + bars['close'].shift(1)) / 2
-    bars.loc[:1, 'ha_open'] = bars['open'].values[0]
-    for x in range(2):
-        bars.loc[1:, 'ha_open'] = (
-            (bars['ha_open'].shift(1) + bars['ha_close'].shift(1)) / 2)[1:]
+    # ha open
+    bars.loc[0:1, 'ha_open'] = (bars['open'].values[0] + bars['close'].values[0]) / 2
+    prev_open = bars[:1]['ha_open'].values[0]
+    for idx, _ in bars[1:][['ha_open', 'ha_close']].iterrows():
```

Comment:
the .loc line (that's vectorized), so the last commit does not improve performance .

The performance problem lies in iterating through all rows instead of using a vectorized operation, which was used before.

I'm not an expert in heikinashi - but I'm pretty certain it can be done vectorized too. (looks like a simple `.lag() / .shift()` operation at a first look).